### PR TITLE
Add today as option, not default

### PR DESCRIPTION
**Pull request by hughbrace**

### CODE REVIEW

File: app/cells/date_range/date_range.coffee [hunk 1]
```diff
@@ -30,17 +30,21 @@ $(document).on 'uic:domchange', (e) ->
...
                       (val) -> val && val.toString().length > 0
 
     options = _.extend({},
       defaults,
-      _.pick($el.data(), ['dateLimit', 'ranges', 'submitOnChange', 'opens']),
+      _.pick($el.data(), ['dateLimit', 'ranges', 'submitOnChange', 'opens', 'today']),
       { startDate: start_date, endDate: end_date })
 
     options.ranges = _.mapObject(options.ranges, (v, k) -> _.map(v, (v) -> moment(v)))
+    options.ranges = _.extend({}, {'Today': [today(), today()]}, options.ranges) if options.today
```
Comment (hughbrace) [inline app/cells/date_range/date_range.coffee]: Used this way in order to ensure the Today option is first. For some reason, although 'Today' is first in the list, 'Yesterday' is still the default selection. Any ideas?

**Comment by dziemba** [comment]
- Why can't we have the "Today" option everywhere?
- I kept "Yesterday" as the default option on purpose. Any reason to change that?

File: app/cells/date_range/date_range.coffee (hunk 1 above)
Comment (ndreynolds) [inline app/cells/date_range/date_range.coffee]: Looks like "Yesterday" gets defaulted up on lines 33 & 35 where the start and end dates are initialized.  I think that's how it works, anyway.

**Comment by hughbrace** [comment]
I think we didn't want to change the default behaviour in case the 'Today' option didn't work with some of the other reports this component is used for.

**Comment by dziemba** [comment]
> in case the 'Today' option didn't work with some of the other reports this component is used for.

Can you give an example?

**Comment by hughbrace** [comment]
I spoke to @hwartig about it yesterday and he said it was better off not changing the default behaviour in case inputting today's date broke any of the other reports (didn't have a specific example). Seemed simpler to add a small option instead of check if it breaks anything across all of the projects it's used in..

**Comment by dziemba** [comment]
I wouldn't add all the complexity in ui_c if we don't need it. 

IMHO all reports should work with (or at least don't break) with the Today option. We can also easily try that out (just use the "Custom Range" to select today)

File: app/cells/date_range/date_range.coffee [hunk 2]
```diff
@@ -30,17 +30,21 @@ $(document).on 'uic:domchange', (e) ->
     $start_input = $($el.data().start)
     $end_input = $($el.data().end)
 
+
+
```
Comment (RKushnir) [inline app/cells/date_range/date_range.coffee]: Are these lines reserved for anything? :)

File: app/cells/date_range/date_range_cell.rb [hunk 3]
```diff
@@ -15,6 +15,7 @@ class DateRangeCell < FormCellBase
     "Accepts: 'right' (default), 'left', 'center'"
   attribute :submit_on_change, description: 'Whether the enclosing form should be ' \
     'automatically submitted on value change'
+  attribute :today, description: 'Whether to include a today optionn the selection'
```
Comment (ndreynolds) [inline app/cells/date_range/date_range_cell.rb]: Small typo here—if we decide to keep the option, anyway.

**Pull request closed without merging**